iXBRLViewerPlugin/__init__.py

```diff
@@ -143,8 +164,42 @@ def setLaunchIXBRLViewer(self, *args):
 
 def guiRun(cntlr, modelXbrl, attach, *args, **kwargs):
     """ run iXBRL Viewer using GUI interactions for a single instance or testcases """
-    if cntlr.hasGui and cntlr.launchIXBRLViewer.get() and modelXbrl.modelDocument.type in (Type.INLINEXBRL, Type.INLINEXBRLDOCUMENTSET):
-        launchLocalViewer(cntlr, modelXbrl)
+    class iXBRLViewerLocalViewer(LocalViewer):
+        # plugin-specific local file handler
+        def getLocalFile(self, file, relpath, request):
+            _report, _sep, _file = file.partition("/")
+            if file == 'ixbrlviewer.js':
+                return static_file('ixbrlviewer.js', os.path.abspath(os.path.join(os.path.dirname(__file__), "viewer", "dist")))
+            elif _report.isnumeric():  # in reportsFolder folder
+                # check if file is in the current or parent directory (may bve
```

Comment:
I'm not sure what you're trying to say at the end of this comment. Did Thor hijack your keyboard? 😄

Reply:
👨‍🍳